src/runtime/classobj.cpp

```diff
@@ -840,6 +966,78 @@ Box* instanceDelslice(Box* _inst, Box* i, Box* j) {
     }
 }
 
+// Analoguous to CPython's, used for sq_slots.
+static int instance_ass_slice(Box* inst, Py_ssize_t i, Py_ssize_t j, PyObject* value) {
+    static BoxedString* delslice_str, *setslice_str, *delitem_str, *setitem_str;
+    Box* func = NULL, * res = NULL;
+    Box* slice = NULL;
+    Box* begin = boxInt(i);
+    Box* end = boxInt(j);
+
+    if (value == NULL) {
+        delslice_str = getStaticString("__delslice__");
+        func = instance_getattro(inst, delslice_str);
+        if (func == NULL) {
+            if (!PyErr_ExceptionMatches(PyExc_AttributeError))
+                return -1;
+            PyErr_Clear();
+
+            delitem_str = getStaticString("__delitem__");
+            func = instance_getattro(inst, delitem_str);
+            if (func == NULL)
+                return -1;
+
+            slice = static_cast<Box*>(createSlice(begin, end, None));
+            res = runtimeCallCapi(func, ArgPassSpec(1), slice, NULL, NULL, NULL, NULL);
+        } else {
+            if (PyErr_WarnPy3k("in 3.x, __delslice__ has been removed; use __delitem__", 1) < 0) {
+                Py_DECREF(func);
+                return -1;
+            }
+            res = runtimeCallCapi(func, ArgPassSpec(2), begin, end, NULL, NULL, NULL);
+        }
+    } else {
+        setslice_str = getStaticString("__setslice__");
+        func = instance_getattro(inst, setslice_str);
+        if (func == NULL) {
+            if (!PyErr_ExceptionMatches(PyExc_AttributeError))
+                return -1;
+            PyErr_Clear();
+
+            setitem_str = getStaticString("__setitem__");
+            func = instance_getattro(inst, setitem_str);
+            if (func == NULL)
+                return -1;
+
+            slice = static_cast<Box*>(createSlice(begin, end, None));
+            res = runtimeCallCapi(func, ArgPassSpec(2), slice, (Box*)value, NULL, NULL, NULL);
+        } else {
+            if (PyErr_WarnPy3k("in 3.x, __setslice__ has been removed; use __setitem__", 1) < 0) {
+                Py_DECREF(func);
+                return -1;
+            }
+            res = runtimeCallCapi(func, ArgPassSpec(3), begin, end, (Box*)value, NULL, NULL);
+        }
+    }
+
+    Py_DECREF(begin);
+    Py_DECREF(end);
+    Py_DECREF(func);
+
+    if (slice == NULL) {
```

Comment:
It is a mistake that I copied the cpython implementation to here (it use ` arg = Py_BuildValue() ` every path), I think this check can be removed and change to AUTO_DECREF(slice) when I create it.